code/modules/mob/living/carbon/carbon.dm

```diff
@@ -21,6 +21,20 @@
 		qdel(dna)
 	return ..()
 
+/mob/living/carbon/gravity_act()
+	if(gravity_stunning)
+		Weaken(gravity_stunning)
+	var/obj/item/device/flightpack/FP = get_flightpack()
+	if(shoes && istype(shoes, /obj/item/clothing/shoes))
+		var/obj/item/clothing/shoes/S = shoes
+		if(S.negates_gravity())
+			if(prob(70))	//Magboots seriously help.

```

Comment:
Alternative is slowdown as magboots. Shouldn't provide complete and absolute protection as it'd make them even more valuable.